src/ModelObjects/representations.rs

```diff
@@ -1229,7 +1238,22 @@ impl QueryExpression {
             }
             QueryExpression::Parentheses(system) => format!("({})", system.pretty_string()),
             QueryExpression::VarName(name) => name.clone(),
-
+            QueryExpression::State(location_names, bool_exp) => {
+                let mut location_name_iter = location_names.iter();
+                let mut res:String = "".to_string();
+                res.push_str(&location_name_iter.next().expect("The location name cannot be empty, '_' should be used instead").pretty_string());
+                for s in location_name_iter {
+                    res.push_str(&(", ".to_string() + &s.pretty_string()));
+                }
+                let bool_str = match bool_exp {
+                    None => "".to_string(),
+                    _ => bool_exp.as_ref().unwrap().to_string()
+                };
+                format!("[{}]({})", res, bool_str)
+            }
+            QueryExpression::LocName(loc_name) => {
+                format!("Locname({})", loc_name)
+            }
```

Comment:
this pretty string seems overly complicated, is there a reason?

Reply:
We have removed this part of code